src/System.IO.FileSystem/src/System/IO/FindEnumerable.Windows.cs

```diff
@@ -0,0 +1,207 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.IO
+{
+    internal partial class FindEnumerable<T> : IEnumerable<T>
+    {
+        private string _directory;
+        private string _originalUserPath;
+        private bool _recursive;
+        private FindTransform<T> _transform;
+        private FindPredicate _predicate;
+
+        /// <summary>
+        /// Encapsulates a find operation. Will strip trailing separator as FindFile will not take it.
+        /// </summary>
+        /// <param name="directory">The directory to search in.</param>
+        /// <param name="expression">
+        /// The filter. Can contain wildcards, full details can be found at
+        /// <a href="https://msdn.microsoft.com/en-us/library/ff469270.aspx">[MS-FSA] 2.1.4.4 Algorithm for Determining if a FileName Is in an Expression</a>.
+        /// </param>
+        /// <param name="getAlternateName">Returns the alternate (short) file name in the FindResult.AlternateName field if it exists.</param>
+        public FindEnumerable(
+            string directory,
+            bool recursive = false,
+            FindTransform<T> transform = null,
+            FindPredicate predicate = null)
+        {
+            _originalUserPath = directory;
+            _directory = Path.GetFullPath(directory);
+            _recursive = recursive;
+            _predicate = predicate ?? throw new ArgumentNullException(nameof(predicate));
+            _transform = transform ?? throw new ArgumentNullException(nameof(transform));
```

Comment:
I don't understand... if these aren't allowed to be null, why are they optional arguments?

Reply:
Whoops- artifact of earlier iteration. Fixing.